ds3-autogen-java/src/main/java/com/spectralogic/ds3autogen/java/JavaCodeGenerator.java

```diff
@@ -84,15 +80,66 @@ public void generate(
     }
 
     private void generateCommands() throws IOException, TemplateException {
+        generateAllRequests();
+        generateAllModels();
+        generateClient();
+    }
+
+    private void generateAllModels() throws IOException, TemplateException {
+        if (spec.getTypes() == null || spec.getTypes().isEmpty()) {
+            return;
+        }
+        for (final Ds3Type ds3Type : spec.getTypes().values()) {
+            generateModel(ds3Type);
+        }
+    }
+
+    private void generateModel(final Ds3Type ds3Type) throws IOException, TemplateException {
+        final Template modelTmpl = getModelTemplate(ds3Type);
+        final Model model = ModelConverter.toModel(ds3Type, getModelPackage());
+        final Path modelPath = getModelPath(model.getName());
+
+        LOG.info("Getting outputstream for file:" + modelPath.toString());
+
+        try (final OutputStream outStream = fileUtils.getOutputFile(modelPath);
+             final Writer writer = new OutputStreamWriter(outStream)) {
+            modelTmpl.process(model, writer);
+        }
+    }
+
+    private Template getModelTemplate(final Ds3Type ds3Type) throws IOException {
+        if (ds3Type.getEnumConstants() != null && !ds3Type.getEnumConstants().isEmpty()) {
```

Comment:
I feel like we should write a util for this check. I've seen almost the same code in lots of places.